core/descriptors/src/org/jetbrains/kotlin/types/KotlinTypeFactory.kt

```diff
@@ -74,19 +74,30 @@ object KotlinTypeFactory {
         arguments: List<TypeProjection>,
         nullable: Boolean,
         kotlinTypeRefiner: KotlinTypeRefiner? = null
+    ) = simpleType(annotations.toDefaultAttributes(), constructor, arguments, nullable, kotlinTypeRefiner)
```

Comment:
Unfortunately this simpleType function is used throughout the compiler; I've temporarily decided to call a toDefaultAttributes() function here to preserve this simpleType function with annotations argument. Since it is used pretty widely throughout the compiler in different situations, it's not clear if this is a correct solution, especially since it does not translate any attributes.